src/ZendeskApi.Client/ZendeskApi.Client.csproj

```diff
@@ -99,30 +138,16 @@
     <Compile Include="Serialization\ZendeskJsonSerializer.cs" />
   </ItemGroup>
   <ItemGroup>
-    <None Include="packages.config" />
+    <Folder Include="lib\" />
   </ItemGroup>
   <ItemGroup>
-    <ProjectReference Include="..\ZendeskApi.Contracts\ZendeskApi.Contracts.csproj">
-      <Project>{bea9e134-91ff-40d6-9b5d-bd57006dbbc6}</Project>
-      <Name>ZendeskApi.Contracts</Name>
-    </ProjectReference>
+    <None Include="App.config" />
+    <None Include="packages.config" />
   </ItemGroup>
   <Import Project="$(MSBuildToolsPath)\Microsoft.CSharp.targets" />
-  <!-- To modify your build process, add your task inside one of the targets below and uncomment it. 
-       Other similar extension points exist, see Microsoft.Common.targets.
-  <Target Name="BeforeBuild">
-  </Target>
-  <Target Name="AfterBuild">
-  </Target>
-  -->
-  <Import Project="$(SolutionDir)\.nuget\NuGet.targets" Condition="Exists('$(SolutionDir)\.nuget\NuGet.targets')" />
-  <!-- To modify your build process, add your task inside one of the targets below and uncomment it. 
-       Other similar extension points exist, see Microsoft.Common.targets.
-  <Target Name="BeforeBuild">
-  </Target>
-  -->
-  <Target Name="AfterBuild">
-    <Copy SourceFiles="$(OutputPath)$(AssemblyName).dll" DestinationFolder="$(OutputPath)..\..\lib\net45" ContinueOnError="true" Condition=" '$(Configuration)|$(Platform)|$(TargetFrameworkVersion)' == 'Release|AnyCPU|v4.5' " />
```

Comment:
Hi Kate

I just made another commit to our fork that addresses this, and the need
for an extra lib directory goes away.  A quick summary: I found that while
building the project via commandline, warnings were thrown around the
.net40 System.Runtime references, while building for .net45.  I addressed
this with conditional logic in the csproj file. The nuspec file was updated
to not look in the lib directory. Since the outputPath is /bin, I simply
made the nuspec file smart enough to look in this directory and pull in
only the files we cared about.  This prevented me from having to add a copy
target to the csproj file.  In addition, I created a new "clean" target
that wipes out the /bin directory before a build. This was extremely
helpful in helping me debug issues and ensure the build was from scratch.

Check it out and let me know if you have any concerns or questions.  Do I
need to issue another pull request since I made another commit to our
repo?  First time using Git.

Also- I'm sure this can be improved upon, however for the quick
convenience, here is a single command that builds the project for both
.net40 and .net45, and then places a nuget package in the root of the
solution directory.  This is run at the solution directory level:

msbuild /p:Configuration=Release ZendeskApiClient.sln /t:clean;build &&
msbuild /p:Configuration=Release40 && nuget pack
src\ZendeskApi.Client\ZendeskApi.Client.nuspec

Patrick

On Tue, Mar 3, 2015 at 5:55 AM, kate-r <notifications@github.com> wrote:

> In src/ZendeskApi.Client/ZendeskApi.Client.csproj
> <https://github.com/justeat/ZendeskApiClient/pull/13#discussion_r25678005>
> :
>
> >    </ItemGroup>
> >    <Import Project="$(MSBuildToolsPath)\Microsoft.CSharp.targets" />
> > -  <!-- To modify your build process, add your task inside one of the targets below and uncomment it.
> > -       Other similar extension points exist, see Microsoft.Common.targets.
> > -  <Target Name="BeforeBuild">
> > -  </Target>
> > -  <Target Name="AfterBuild">
> > -  </Target>
> > -  -->
> > -  <Import Project="$(SolutionDir)\.nuget\NuGet.targets" Condition="Exists('$(SolutionDir)\.nuget\NuGet.targets')" />
> > -  <!-- To modify your build process, add your task inside one of the targets below and uncomment it.
> > -       Other similar extension points exist, see Microsoft.Common.targets.
> > -  <Target Name="BeforeBuild">
> > -  </Target>
> > -  -->
> > -  <Target Name="AfterBuild">
> > -    <Copy SourceFiles="$(OutputPath)$(AssemblyName).dll" DestinationFolder="$(OutputPath)..\..\lib\net45" ContinueOnError="true" Condition=" '$(Configuration)|$(Platform)|$(TargetFrameworkVersion)' == 'Release|AnyCPU|v4.5' " />
>
> These two after build copy steps, to copy the dlls from the release bin,
> to the \ZendeskApiClient\src\ZendeskApi.Client\lib\net45 folder are used by
> our CI Build to make a nuget package from. Could you put the two copy
> targets back.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/justeat/ZendeskApiClient/pull/13/files#r25678005>.
>